### ISSUE TITLE
Error when changing location or wallpaper

### ISSUE THREAD

**Issue by nukmuk** [t1m0thyj/WinDynamicDesktop#599]
- WDD Version: 5.5.0
- OS Version: Windows 11 23H2

Steps to Reproduce:

1. Set some location such as Finland
2. Choose theme with more than a few wallpapers such as Mojave (some themes seem to work, some don't, even though they both have 16 wallpapers)

Are you using the Microsoft Store version of WDD?: Yes

![image](https://github.com/user-attachments/assets/c6d40b66-5eae-4ed5-8675-a188bc774d09)

```
[2024-07-21 21:23:34.954] System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at WinDynamicDesktop.SolarScheduler.CalcNextUpdateTime(SolarData data, DisplayEvent e)
   at WinDynamicDesktop.EventScheduler.Run(Boolean forceImageUpdate, DisplayEvent overrideEvent)
   at WinDynamicDesktop.ThemeDialog.ApplySelectedTheme()
   at WinDynamicDesktop.ThemeDialog.applyButton_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)```

**Comment by t1m0thyj** [comment]
Thanks for reporting this! I believe I've found the problem for places like northern Finland where the sun is up all day for part of the summer.

Could you try this test build to see if the issue is fixed? https://github.com/t1m0thyj/WinDynamicDesktop/actions/runs/10122987874

**Comment by nukmuk** [comment]
I've been using the build you provided for a few days now and the issue seems to be fixed, thanks!

**Comment by t1m0thyj** [comment]
Fixed in WinDD 5.6.0
